api/v1alpha1/hostedcluster_types.go

```diff
@@ -318,6 +327,20 @@ type NodePortPublishingStrategy struct {
 	Port int32 `json:"port,omitempty"`
 }
 
+// LoadBalancerPublishingStrategy specifies setting used to expose a service as a LoadBalancer.
+type LoadBalancerPublishingStrategy struct {
+	// Hostname is the DNS name that will be created via external-dns pointing to the LoadBalancer.
```

Comment:
Nit: maybe omit the external-dns reference on the API docs as this is an implementation detail?